appstore/locker.py

```diff
@@ -113,3 +113,20 @@ def app_locker(app_uuid):
             db.session.delete(entry)
             db.session.commit()
         return '', 204
+
+
+@api.route("/locker/by_token/<user_token>")
+def app_locker_by_token(user_token):
```

Comment:
We should probably restrict it in some manner — if this is publicly accessible the user is substantially deanonymised.

Reply:
@jwise, @Katharine I don't think we can restrict this because requests here are coming from the public api handler `/v1/user/pins/ID` and everything it gets is `X-User-Token`. The only way I could imagine is to somehow validate that requester's IP address matches timeline-sync server's.

Reply:
Hmm, how about a shared secret between get_locker_info and appstore/locker?  One quick-and-dirty approach: in `appstore/settings.py`, add a `config` entry for `INTERNAL_SECRET`, and check for that in an `Authorization: Bearer <secret>` header.  I'll take a crack at that, though the current docker-compose setup is more than a little out of date, as you noted...

Then, on the timeline-sync side, we'd simply do the opposite.

Reply:
Ok, I've prototyped this in the jwise/timeline branch, commit 6ace2d8a257acb369895e05c502ff5c7cfab0a26 ... but of course since I don't have it running in my rws-compose setup, I haven't tested it, so take it with a grain of salt until I get the pieces put together :-)